### PR TITLE
feat: allow filtering for null

### LINKED ISSUE #757: Filter by `null` values
It should be possible to filter for `null` valued fields.

I just checked on the Loculus Ebola instance:

### For GET requests it seems to work:
`GET /sample/details?accessionVersion=LOC_0001D2F.1&ncbi_is_lab_host=` yields a result.
Taken from this swagger request:
grafik

### For POST it does not

```
POST /sample/details
{
  "accessionVersion": "LOC_0001D2F.1",
  "ncbi_is_lab_host": null
}
``` 

yields no result.

```
POST /sample/details
{
  "accessionVersion": "LOC_0001D2F.1"
}
``` 
yields the same result as the GET request.

### Expected behavior

* Both GET and POST should return the same results.
* For both methods it should be possible to filter by `null`. This should be possible for every sequence filter field that LAPIS offers.

### PR DESCRIPTION
**Pull request by JonasKellerer**
resolves #757

## Description
Allows all filters in GET and POST requests to be null (except pango lineage)

## PR Checklist
- [ ] All necessary documentation has been adapted.
- [x] The implemented feature is covered by an appropriate test.

### CODE REVIEW

**Comment by JonasKellerer** [comment]
Currrently this PR is waiting on SILO to handle null values in float filters.

**Comment by JonasKellerer** [comment]
@fengelniederhammer Do we want to explicitly show in Swagger that null is a valid filter value?

**Comment by fengelniederhammer** [comment]
> @fengelniederhammer Do we want to explicitly show in Swagger that null is a valid filter value?

Would be good, does Swagger have good support for nullable types? Or is it that "not required" means "nullable" in Swagger?

File: lapis2/src/test/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapperTest.kt
Comment (fengelniederhammer) [inline lapis2/src/test/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapperTest.kt:1]: Let's also add a test for a `dateFrom` and `dateTo` with explicitly set `null`s? I think `getAsDate` in the implementation needs some null handling, too.

File: lapis2/src/test/kotlin/org/genspectrum/lapis/silo/SiloQueryTest.kt [hunk 1]
```diff
@@ -361,6 +361,16 @@ class SiloQueryTest {
...
                 """,
                 ),
+                Arguments.of(
+                    StringEquals("theColumn", null),
+                    """
+                {
+                    "type": "StringEquals",
+                    "column": "theColumn",
+                    "value": null
+                }
```
Comment (fengelniederhammer) [inline lapis2/src/test/kotlin/org/genspectrum/lapis/silo/SiloQueryTest.kt:371]:
```suggestion
                    {
                        "type": "StringEquals",
                        "column": "theColumn",
                        "value": null
                    }
```

Comment (fengelniederhammer) [reply lapis2/src/test/kotlin/org/genspectrum/lapis/silo/SiloQueryTest.kt:371]: (Actually also the indent of most other JSONs here is wrong - at least for my taste)

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt:1]: Line 243 shows a warning:
```
Type mismatch.
Required: CharSequence
Found: String?
```

**Review by fengelniederhammer — CHANGES_REQUESTED** [review]
Looks good :+1:

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt
Comment (JonasKellerer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt:1]: Fixed with null handling.

File: lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt [hunk 2]
```diff
@@ -308,7 +322,7 @@ class SiloFilterExpressionMapper(
         val value = extractSingleFilterValue(values, originalKey)
 
         try {
-            return value.toInt()
+            return value?.toInt()
```
Comment (fengelniederhammer) [inline lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt:325]: Does this also need a "is blank" handling?
`/sample/aggregated?ageFrom=` returns a bad request with `"ageFrom '' is not a valid integer: For input string: \"\""`, while a POST with `{ "age": null}` is ok.

Comment (fengelniederhammer) [reply lapis2/src/main/kotlin/org/genspectrum/lapis/model/SiloFilterExpressionMapper.kt:325]: (Same for floats)

**Review by fengelniederhammer — APPROVED** [review]
